File: morph-common/src/main/java/communication/dao/Job.java
package communication.dao;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.ToString;

import java.util.List;

@Data
@ToString(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class Job {
    private Integer jobId;
    private String name;
    private Integer nthreads;
    private String CPU;
    private String RAM;
    private String startTime;
    private String Duration;
    private Boolean isRunning;
    private Integer nEvents;
    private Double minProcessTime;
    private Double maxProcessTime;
    private Double meanProcessTime;
    private Double latency;
    private Double throughput;
    private Integer ncore;
    private Operator[] operators;
    private OverallTimeBreakdown overallTimeBreakdown;
    private SchedulerTimeBreakdown schedulerTimeBreakdown;
    private List<Double> periodicalThroughput;
    private List<Double> periodicalLatency;
}

File: morph-web/dashboard/src/app/model/Job.ts
import {Operator} from "./Operator";
import {TotalTimeBreakdown} from "./TotalTimeBreakdown";
import {OverallTimeBreakdown} from "./OverallTimeBreakdown";

export interface Job {
  jobId: string;
  name: string;
  nthreads: string;
  cpu: string;
  ram: string;
  startTime: string;
  duration: string;
  isRunning: boolean;
  nevents: number;
  minProcessTime: number;
  maxProcessTime: number;
  meanProcessTime: number;
  latency: number;
  throughput: number;
  ncore: number;
  operators: Operator[];
  totalTimeBreakdown: TotalTimeBreakdown;
  schedulerTimeBreakdown: OverallTimeBreakdown;
  periodicalThroughput: number[];
  periodicalLatency: number[];
}
